heareval/predictions/task_predictions.py

```diff
@@ -146,32 +204,62 @@ def __init__(
         self.labels = pickle.load(
             open(embedding_path.joinpath(f"{split_name}.target-labels.pkl"), "rb")
         )
+        # Only used for event-based prediction
```

Comment:
Same here -- different classes of DatasetLoaders for either event or prediction types?

Reply:
Then we can avoid all the logic in these functions that is looking for either event or scene.

Reply:
I think I was able to make this clean by having a dataset metadata dict. It was very simple to change the code to use this. The only downside really is a bit less type-safety and things are a little less explicit.

Reply:
hmm ya. interesting -- on one hand it creates some flexible down the road if we want to pass other types of metadata depending on the task type. It's not ideal have to pass around the empty dict for the scene stuff. I can't think of a better approach atm though, this seems to be working pretty well.

Reply:
Yes. The metadata dict looks nice and it does give more flexibility.